### ISSUE TITLE
httpPath clarity

### ISSUE THREAD

**Issue by amoose136** [gershnik/wsdd-native#7]
Hi I've been trying to get network discovery of a simple http server running on a single board computer (SBC) working on Windows and this looks like it will almost do what I want. I tried it out and it does make the SBC show up under the "Computer" section of the networking tab of File Explorer. What doesn't work, however, is double clicking the device. I'd like it to go to an address (either a fixed ip address or hostname.local address that avahi can resolve). I can't find a way to get that part configured. Currently, double clicking the SBC from the network tab gives an error that says:

"Windows cannot access \\\\`the hostname I picked`" 

In the details dropdown it says "Error code: 0x80070035 The network path was not found." 

I checked the source code and came across something that didn't make sense to me in config.h. It seems to me that httpPath ought to setting where double clicking the device goes but this is set to m_strUuid.
````
    auto endpointIdentifier() const -> const sys_string &   { return m_urnUuid; }
    auto httpPath() const -> const sys_string &             { return m_strUuid; }
````

To me the above m_strUuid doesn't seem like the right token to go there but I also don't really understand how all this works.

So I have two questions:

1. Is there a way to configure which section a device will appear under? (IE not just "Computer")
2. Is there a way to configure where double clicking the device will go?

**Comment by gershnik** [comment]
Do you have Samba running on the target device?

(`wsdd-native` only provides the "name" of the device to Windows to show in Explorer. It doesn't do the actual file sharing - for that you need SMB sharing service - which is usually Samba.)

Currently `wsdd-native` marks the device it exposes as a "computer" which means that Explorer will try to access it as `\\hostname` via SMB protocol. If your target device isn't an SMB server but something else then it is not going to work with the current version. 

It might be possible to add support for other device types but I would need to understand a whole lot more about what kind of device this is and how it is supposed to be accessed. 

(`endpointIdentifier` is just a unique ID in WS-Discovery protocol. It has nothing to do with hostnames etc.)

**Comment by amoose136** [comment]
For this application, it's not a Samba server. It's an HTTP server.  What I'm trying to do is almost exactly what the Octoprint discovery plugin does. https://github.com/OctoPrint/OctoPrint/tree/master/src/octoprint/plugins/discovery
IE the advertised schema is something like what's in:
https://github.com/OctoPrint/OctoPrint/blob/master/src/octoprint/plugins/discovery/templates/discovery.xml.jinja2

**Comment by gershnik** [comment]
Oh I see. Funny, when I was writing this I thought about whether to make the device type configurable and decided "nah, too much work, who'd need that?". 🙂

Making it configurable is indeed quite a bit of work but if you are looking at hacking the source code yourself then doing what you want is quite easy. See 
```
void fill(const GetResponse & val, XmlNode & bodyNode, const Namespaces & ns) 
```
at line 167 in `wsd_server.cpp`. You can reconfigure the XML response there to fit your needs. Hopefully the code is reasonably clear.

**Comment by gershnik** [comment]
@amoose136 
Well you got me curious with this issue so I went ahead and implemented support for custom metadata that should (hopefully) take care of your scenario. It is on master branch but hasn't been released yet. Details are at [config/metadata/README.md](https://github.com/gershnik/wsdd-native/blob/master/config/metadata/README.md).

This [example](https://github.com/gershnik/wsdd-native/blob/master/config/metadata/other.xml) should work for your scenario.

If you are still interested in this, please give it a try and let me know if it works!

Thank you!

**Comment by gershnik** [comment]
This has now been released in version 1.10

**Comment by amoose136** [comment]
Thanks, I'll be sure to try it out soon.
